apps/jupyter/extensions/notebook-runner/src/iframe-api.ts

```diff
@@ -407,6 +407,67 @@ export class EmbeddedPythonCallbacks {
       throw new Error(`Unsupported content format: ${file.format}`);
     }
   }
+
+  private async writeFolderContents(
+    basePath: string,
+    files: Map<string, Blob>,
+  ): Promise<void> {
+    for (const [relativePath, blob] of files.entries()) {
+      const path = `${basePath}/${relativePath}`;
+      const pathParts = relativePath.split("/");
+
+      if (pathParts.length > 1) {
```

Comment:
In places like this a small comment like `// Create all folders along the path` can help reading :)